File: problems/010/010.f90
PROGRAM F010

    ! Copyright 2021 Melwyn Francis Carlo

    IMPLICIT NONE

    INTEGER, PARAMETER :: N = 2E+6
    INTEGER :: I, J
    INTEGER (KIND=8) :: PRIMES_SUM
    LOGICAL, DIMENSION(N+1) :: INTEGERS_LIST

    ! USING THE ALGORITHM OF SIEVE OF EROSTHENES
    INTEGERS_LIST = .TRUE.
    I = 2
    DO WHILE ((I*I) <= N)
        IF (INTEGERS_LIST(I)) THEN
            J = I * I
            DO WHILE (J <= N)
                INTEGERS_LIST(J) = .FALSE.
                J = J + I
            END DO
        END IF
        I = I + 1
    END DO
    PRIMES_SUM = 0
    DO I = 2, N
        IF (INTEGERS_LIST(I)) THEN
            PRIMES_SUM = PRIMES_SUM + I
        END IF
    END DO

    PRINT ('(I0)'), PRIMES_SUM

END PROGRAM F010

File: problems/010/010.c
#include <stdio.h>
#include <stdlib.h>

/* Copyright 2021 Melwyn Francis Carlo */

int main() {
    int n = 2E+6;
    // Using the algorithm of Sieve of Eratosthenes
    long int *integers_list = (long int*)calloc(n+1, sizeof(long int));
    for (long int i = 2; (i*i) <= n; i++) {
        if (integers_list[i] == 0) {
            for (long int j = (i*i); j <= n; j += i)
                integers_list[j] = 1;
        }
    }
    long int primes_sum = 0;
    long int i = 2;
    while (i < n) {
        if (integers_list[i] == 0)
            primes_sum += i;
        i += 1;
    }
    printf("\n%ld\n\n", primes_sum);
    free(integers_list);
    return 0;
}
